src/tests/test_magic.py

```diff
@@ -195,6 +195,162 @@ def test_persist_bare(ip):
     assert result.error_in_exec
 
 
+def query_and_save_as_dataframe(ip, table_name, limit):
```

Comment:
I think the name might confuse since we are running a specific query on a given table (`SELECT * FROM table`).
Maybe we can change it to something like `get_table_rows_as_dataframe` (or a better one) and initiate `limit` with a default number, so the purpose of this function would be to "return X rows from the given table as dataframe."

Question - I see you are mostly fetching only 1 line from the table. Does it matter for the tests? If not, let's put a default number of 5, if it matters maybe the default number should be 1.

this way, we can turn this line:

```
saved_df_name = query_and_save_as_dataframe(ip, table_name=test_table, limit=1)
```

to 

```
saved_df_name = get_table_rows_as_dataframe(ip, table_name=test_table)
```

which might be easier to read, use and maintain.

what do you think?

Reply:
yes, but do we actually test these different lengths? I think in most of the cases I've seen, we test the actual values, so the limit doesn't make much difference.

Reply:
Oh sorry I think we did not test the length, but the actual value

I can rename the function to `get_table_rows_as_dataframe` and remove the limit part, although we will always use same dataframe to store by `--persist` and -`-persist-replace`